### PR TITLE
test: added test for gradient clipping in fsdpX

### PR DESCRIPTION
**Pull request by fromm-m**
# What does this PR do?

This PR includes tests for FSDPXGradientClipper.

## General Changes
* ..

## Breaking Changes
* .. 

## Checklist before submitting final PR
- [x] My PR is minimal and addresses one issue in isolation
- [x] I have merged the latest version of the target branch into this feature branch
- [x] I have reviewed my own code w.r.t. correct implementation, missing type hints, proper documentation, etc.
- [x] I have run a sample config for model training
- [x] I have checked that all tests run through (`python tests/tests.py`)
- [x] I have updated the internal changelog (`CHANGELOG_DEV.md`)

### CODE REVIEW

File: tests/test_gradient_clipping.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    max_norm = 1.0
+    norm_type = GradientClippingMode.P2_NORM
+    clipper = FSDP1GradientClipper(wrapped_model=mock_model, max_norm=max_norm, norm_type=norm_type)
+
+    # Call clip_gradients
+    clipper.clip_gradients()
+
+    # Verify that clip_grad_norm_ was called with the correct arguments
+    mock_model.clip_grad_norm_.assert_called_once_with(max_norm=max_norm, norm_type=norm_type.value)
+    # Note: The actual norm returned depends on the mock's return value, which isn't tested here
```
Comment (le1nux) [inline tests/test_gradient_clipping.py:45]: Can we also check here if the gradients are scaled down to the expected values, like we did for the FSDP2 tests?

**Review by le1nux — CHANGES_REQUESTED** [review]
One minor comment, otherwise looks great!

**Review by le1nux — APPROVED** [review]
LGTM :)
